pyproject.toml

```diff
@@ -27,8 +27,34 @@ extend_skip = ["setup.py", "docs/source/conf.py"]
 [tool.black]
 line_length = 119
 skip_string_normalization = true
-required_version = "19.10b0"  # recongized by future versions, disallows to reformat code with incompatible versions
-
+# major year version is stable, see details in
+# https://black.readthedocs.io/en/stable/the_black_code_style/index.html
+required_version = "24"
+target-version = ['py310', 'py311', 'py312']
```

Comment:
There are four places we are referencing black version now, in the action, in the toml and the pre commit, requirements txt. Can we somehow have a unified place that sets version for all three. 

If not easy, we need to write somewhere that all 4 places need to be updated simultaneously.

Reply:
For actions (pre-commit, GitHub) we need to specify action version, but not black. It is not possible to do this in one place. But `required_version="24"` in this file **guarantees the consistency**: 
any action with black will fail if it `black` major version is different from `required_version`

Reply:
It is also beneficial in cases when the user locally has a different black version (not from the requirements). Instead of incorrectly reformatting the files, it will fail.